src/components/share-button/share-button.component.ts

```diff
@@ -105,24 +73,25 @@ export class ShareButtonComponent implements AfterViewInit {
                 this.popUpClosed.emit(this.button.provider);
             }
         }, 200);
-    }
+    };
 
+    shareButton() {
+        this.btn = this.renderer.createElement(this.self, 'button');
+        this.renderer.setElementProperty(this.btn, 'innerHTML', this.button.template);
+        this.renderer.setElementAttribute(this.btn, 'class', this.button.classes);
+        this.renderer.setElementProperty(this.btn, 'onclick', this.share);
+    }
 
-    nFormatter(num, digits) {
-        let si = [
-            { value: 1E18, symbol: 'E' },
-            { value: 1E15, symbol: 'P' },
-            { value: 1E12, symbol: 'T' },
-            { value: 1E9, symbol: 'G' },
-            { value: 1E6, symbol: 'M' },
-            { value: 1E3, symbol: 'K' }
-        ], rx = /\.0+$|(\.[0-9]*[1-9])0+$/, i;
-        for (i = 0; i < si.length; i++) {
-            if (num >= si[i].value) {
-                return (num / si[i].value).toFixed(digits).replace(rx, '$1') + si[i].symbol;
-            }
-        }
-        return num.toFixed(digits).replace(rx, '$1');
+    shareCount() {
+        this.sbService.count(this.button.provider, this.url)
+            .subscribe(shareCount => {
+                if (shareCount) {
+                    let counter = this.renderer.createElement(this.self, 'span');
```

Comment:
Well, The plan was like this from the beginning, I tried 3 different ways before using `renderer`

 - if you use `async` pipe in the view it will send +1000 requests and will get the blocked.

 - If you make a child component e.g. `<share-count>` and use `[count]=sbService.count(button.provider, url) | async` you will also get the same result

 - if you assign the count result value to a variable and bind it to the view, then you have to remove the `ChangeDetectionStrategy.OnPush` because it won't update

By using the `renderer` I was able to add the count element only when the count is enabled, which gave the expected behavior, this was the only solution I found, other solution could be using a single state for the whole package and this requires some big changes.

So since I used the `renderer` once I made the changes of the `"avoid DOM manipulation" comment` to make it one style you know...

On the other hand I do not have experience in writing tests yet, so I'm not aware of the difficulties on that area

Have you been able to get it work using the template?